pennylane/qchem/hartree_fock.py

```diff
@@ -189,6 +191,7 @@ def nuclear_energy(charges, r):
     Args:
         charges (list[int]): nuclear charges in atomic units
         r (array[float]): nuclear positions
+        argnum (int, list(int), None): differentiability of coords, coeffs, and alpha (in that order)
```

Comment:
Docstring not great. Maybe something like that in `qml.grad`?
 
 > Which argument(s) to take the gradient with respect to. 0 for coordinates, 1 for coefficients, and 2 for alpha.
 
 And add an example?